src/memory_pool_tracking.c

```diff
@@ -0,0 +1,93 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include "memory_pool_internal.h"
+#include "memory_provider_internal.h"
+#include "memory_tracker.h"
+
+#include <umf/memory_pool.h>
+
+#include <assert.h>
+#include <stdlib.h>
+
+enum umf_result_t umfPoolCreate(const struct umf_memory_pool_ops_t *ops,
+                                umf_memory_provider_handle_t provider,
+                                void *params,
+                                umf_memory_pool_handle_t *hPool) {
+    if (!provider) {
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+
+    enum umf_result_t ret = UMF_RESULT_SUCCESS;
+    umf_memory_pool_handle_t pool = malloc(sizeof(struct umf_memory_pool_t));
+    if (!pool) {
+        return UMF_RESULT_ERROR_OUT_OF_HOST_MEMORY;
+    }
+
+    assert(ops->version == UMF_VERSION_CURRENT);
+
+    pool->provider = calloc(1, sizeof(umf_memory_provider_handle_t));
```

Comment:
Hm, I don't think we need this calloc anymore, this was needed when we had an array of providers, but now just calling `umfTrackingMemoryProviderCreate` on `pool->provider` should be enough.

Reply:
Right, done.